twisted/conch/test/test_default.py

```diff
@@ -147,23 +169,129 @@ def test_getPrivateKeyPassphrase(self):
+        user succeeds.
+
+        """
+        class FakeTransport:
+            def __init__(self, host):
+                self.transport = self
+                self.host = host
+            def getPeer(self):
+                return self
+
+        options = ConchOptions()
+        client = SSHUserAuthClient(b"user",  options, None)
+        client.transport = FakeTransport("127.0.0.1")
+
+        def getpass(prompt):
+            self.assertEqual(prompt, "user@127.0.0.1's password: ")
+            return 'bad password'
+
+        self.patch(default.getpass, 'getpass', getpass)
+        d = client.getPassword()
+        d.addCallback(self.assertEqual, b'bad password')
+        return d
+
+    test_getPassword.skip = windowsSkip or ttySkip
+
+
+    def test_getPasswordPrompt(self):
+        """
+        Verify that the getPassword function in L{SSHUserAuthClient}
+        works, when prompt is sent, and fetching the password from the
+        user succeeds.
+        """
+        options = ConchOptions()
+        client = SSHUserAuthClient(b"user",  options, None)
+        prompt = b"Give up your password"
+
+        def getpass(p):
+            self.assertEqual(p, nativeString(prompt))
+            return 'bad password'
+
+        self.patch(default.getpass, 'getpass', getpass)
+        d = client.getPassword(prompt)
+        d.addCallback(self.assertEqual, b'bad password')
+        return d
+
+    test_getPasswordPrompt.skip = windowsSkip or ttySkip
+
+
+    def test_getPasswordConchError(self):
+        """
+        Verify that the getPassword function in L{SSHUserAuthClient}
+        fails it's deferred if the underlying _getPassword function
+        raises a ConchError
+        """
+        options = ConchOptions()
+        client = SSHUserAuthClient(b"user",  options, None)
+
+        def getpass(prompt):
+            raise KeyboardInterrupt("User pressed CTRL-C")
+
+        self.patch(default.getpass, 'getpass', getpass)
+        stdout, stdin = sys.stdout, sys.stdin
+        d = client.getPassword(b'?')
+        @d.addErrback
+        def check_sys(fail):
+            self.assertEqual(
+                [stdout, stdin], [sys.stdout, sys.stdin])
+            return fail
+        self.assertFailure(d, ConchError)
+
+    test_getPasswordConchError.skip = windowsSkip or ttySkip
+
+
+    def test_getGenericAnswers(self):
+        """
+        Verify that the getGenericAnswers function in L{SSHUserAuthClient}
+        works.
+        """
+        options = ConchOptions()
+        client = SSHUserAuthClient(b"user",  options, None)
+
+        def getpass(prompt):
+            self.assertEqual(prompt, "pass prompt")
+            return "getpass"
+
+        self.patch(default.getpass, 'getpass', getpass)
+
+        def raw_input(prompt):
+            self.assertEqual(prompt, "raw_input prompt")
+            return "raw_input"
+
+        self.patch(default, 'raw_input', raw_input)
+        d =client.getGenericAnswers(
```

Comment:
twistedchecker, where art thou: there should be a second space after the equals sign